odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1064 @@
+        )
+
+        # read the default many2one and deactivate the name_get
+        read_values = self.discussion_1.read(['message_properties'], load=None)[0]['message_properties']
+        self.assertEqual(
+            read_values[0]['default'],
+            self.partner.id,
+            msg='If the name_get is deactivate, it should not return the display name',
+        )
+
+        message = self.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'author': self.user.id,
+            'discussion': self.discussion_1.id,
+        })
+
+        properties = message.read(['custom_properties'])[0]['custom_properties']
+        self.assertEqual(properties[0]['value'], (self.partner.id, self.partner.display_name))
+
+        self.assertEqual(message.custom_properties[0]['value'], self.partner.id)
+
+        # give a default value and a value for a many2one
+        # the default value must be ignored
+        partner_2 = self.env['res.partner'].create({'name': 'Partner 2'})
+        property_definition = self.discussion_1.read(['message_properties'])[0]['message_properties']
+        property_definition[0]['value'] = (partner_2.id, 'Alice')
+        message = self.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'author': self.user.id,
+            'discussion': self.discussion_1.id,
+            'custom_properties': property_definition,
+        })
+        self.assertEqual(
+            message.custom_properties[0]['value'],
+            partner_2.id,
+            msg='Should not take the default value',
+        )
+
+    def test_properties_field_read(self):
+        """Test the behavior of the read method.
+
+        In comparison with a simple "record.properties", the read method should not
+        record a recordset for the many2one, but a tuple with the record id and
+        the record name_get.
+        """
+        properties_values = (self.message_1 | self.message_3).read(['custom_properties'])
+
+        self.assertEqual(len(properties_values), 2)
+
+        properties_message_1 = properties_values[0]['custom_properties']
+        properties_message_3 = properties_values[1]['custom_properties']
+
+        self.assertTrue(isinstance(properties_message_1, list))
+        self.assertTrue(isinstance(properties_message_3, list))
+
+        self.assertEqual(len(properties_message_1), 2, msg="Message 1 has 2 properties")
+        self.assertEqual(len(properties_message_3), 1, msg="Message 3 has 1 property")
+
+        self.assertEqual(
+            properties_message_1[0]['id'], 'name',
+            msg='First message 1 property should be "name"')
+        self.assertEqual(
+            properties_message_1[1]['id'], 'partner_id',
+            msg='First message 1 property should be "partner_id"')
+        self.assertEqual(
+            properties_message_3[0]['id'], 'state',
+            msg='First message 3 property should be "state"')
```

Comment:
There seems to be some mismatch in the explanation strings.
That or I'm very lost ^^